optuna/integration/pytorch_distributed.py

```diff
@@ -21,18 +22,18 @@
     import torch
     import torch.distributed as dist
 
-
 if TYPE_CHECKING:
     from typing_extensions import ParamSpec
 
     _T = TypeVar("_T")
     _P = ParamSpec("_P")
 
-
 _suggest_deprecated_msg = (
     "Use :func:`~optuna.integration.TorchDistributedTrial.suggest_float` instead."
 )
 
+_g_pg: List[Optional["torch.distributed.ProcessGroup"]] = [None]
```

Comment:
We cannot.

Since the `TorchDistributedTrial` is created multiple times during training, we need a global variable to store `gloo` backend by default. Because process group is an application scoped variable, and `TorchDistributedTrial` is per-trial (or objective local ) variable.

See  https://github.com/optuna/optuna-examples/blob/main/pytorch/pytorch_distributed_simple.py#L100

Reply:
In a distributed optimization setup, even global variables are not shared in memory across processes, right? I think it is equivalent to storing information to properly identify the process group for each `TorchDistributedTrial`.